src/index/IndexBuilderMain.cpp

```diff
@@ -247,10 +247,14 @@ int main(int argc, char** argv) {
     }
   }
 
+  // If no text index name was specified, take the part of the wordsfile after
+  // the last slash.
   if (textIndexName.size() == 0 && wordsfile.size() > 0) {
     textIndexName = ad_utility::getLastPartOfString(wordsfile, '/');
   }
 
+  // If no index name was specified, take the part of the input file name after
+  // the last slash.
   if (kbIndexName.size() == 0) {
```

Comment:
Also use `empty()` and  `&&`for the two conditions here (couldn't create a suggestion here).